src/sentry/search/django/backend.py

```diff
@@ -139,12 +139,33 @@ def _build_queryset(self, project, query=None, status=None, tags=None,
             )
 
         if tags:
-            matches = self._tags_to_filter(project, tags)
-            if not matches:
+            filter_tags, exclude_tags = {}, {}
+            for k, v in six.iteritems(tags):
+                if k[0] == '-':
+                    exclude_tags[k[1:]] = v
+                else:
+                    filter_tags[k] = v
+            # Don't allow the same tag to be both excluded and included
+            common_tags = (k for k in filter_tags if k in exclude_tags)
```

Comment:
You are right, I wasn't paying attention either. The convention makes sense, as long as we stick to the convention. Pushing a fix.